app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -0,0 +1,91 @@
+package com.kamron.pogoiv;
+
+import android.annotation.TargetApi;
+import android.graphics.Bitmap;
+import android.graphics.PixelFormat;
+import android.hardware.display.DisplayManager;
+import android.media.Image;
+import android.media.ImageReader;
+import android.media.projection.MediaProjection;
+import android.os.Build;
+import android.util.DisplayMetrics;
+import android.widget.Toast;
+
+import java.nio.ByteBuffer;
+
+import timber.log.Timber;
+
+/**
+ * Created by Sarav on 8/27/2016.
+ */
+public class ScreenGrabber {
+
+    private static ScreenGrabber instance = null;
+    private ImageReader mImageReader;
+    private MediaProjection mProjection = null;
+    private DisplayMetrics rawDisplayMetrics;
+    private DisplayMetrics displayMetrics;
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private ScreenGrabber(MediaProjection mediaProjection, DisplayMetrics raw, DisplayMetrics display) {
+        rawDisplayMetrics = raw;
+        displayMetrics = display;
+        mProjection = mediaProjection;
+        mImageReader = ImageReader.newInstance(rawDisplayMetrics.widthPixels, rawDisplayMetrics.heightPixels, PixelFormat.RGBA_8888, 2);
+        mProjection.createVirtualDisplay("screen-mirror", rawDisplayMetrics.widthPixels, rawDisplayMetrics.heightPixels, rawDisplayMetrics.densityDpi, DisplayManager.VIRTUAL_DISPLAY_FLAG_PUBLIC, mImageReader.getSurface(), null, null);
+    }
+
+    public static ScreenGrabber init(MediaProjection mediaProjection, DisplayMetrics raw, DisplayMetrics display) {
+        if (instance == null) {
+            instance = new ScreenGrabber(mediaProjection, raw, display);
+        }
+        return instance;
+    }
+
+    public void exit() {
+        if (mProjection != null) {
+            mImageReader = null;
+            mProjection.stop();
+            mProjection = null;
+            rawDisplayMetrics = null;
+            displayMetrics = null;
+            instance = null;
```

Comment:
I look at it the other way: If you don't have access to the instance, you don't get to call exit(). If exit() was a static function, then anyone could call it. I obviously have to clear out "instance" from there though since if I don't, then I can't call .init() again.